test/e2e: Move tests into test/e2e/node package

navidshaikh opened the pull request:
#### What type of PR is this?
/kind cleanup
/priority important-soon

#### What this PR does / why we need it:
As part of a reliability working group effort the workstream of marking tests with SIG-identifier attribute has started. https://groups.google.com/g/kubernetes-sig-node/c/dVaNhlgT9CU/m/HXwY6slvAQAJ
Move `test/e2e/examples.go` to `test/e2e/node` and assign sig-node ownership.

#### Which issue(s) this PR fixes:
ref #98326 

#### Special notes for your reviewer:
 
#### Does this PR introduce a user-facing change?
```release-note
NONE
```

navidshaikh commented on test/e2e/framework/framework.go:
```diff
@@ -635,6 +635,11 @@ func KubeDescribe(text string, body func()) bool {
 	return ginkgo.Describe("[k8s.io] "+text, body)
 }
 
+// SIGDescribe is a wrapper function for ginkgo describe. Adds SIG namespacing.
+func SIGDescribe(sig, text string, body func()) bool {
+	return ginkgo.Describe("[sig-"+sig+"] "+text, body)
+}
+
```
Added this following review suggestion from https://github.com/kubernetes/kubernetes/pull/98957 
cc: @liggitt @wojtek-t

wojtek-t commented on test/e2e/gke_node_pools.go:
```diff
@@ -27,7 +27,7 @@ import (
 )
 
-var _ = framework.KubeDescribe("GKE node pools [Feature:GKENodePool]", func() {
+var _ = framework.SIGDescribe("node", "GKE node pools [Feature:GKENodePool]", func() {
```
I would say it's more cluster-lifecycle thing

neolit123 replied:
the current agreement is that GKE test specifics fall under SIG Cloud Provider (`/area gcp`)
SIG CL tries to be product agnostic.

ehashman replied:
Shall we set this to cloud-provider then?

wojtek-t replied:
yeah - ok - cloud-provider is better then

navidshaikh replied:
raised separate PR per sig tag 
 - sig-cloud-provider https://github.com/kubernetes/kubernetes/pull/99086

wojtek-t commented on test/e2e/node/recreate_node.go:
```diff
@@ -49,7 +49,7 @@ func nodeNames(nodes []v1.Node) []string {
 }
 
-var _ = ginkgo.Describe("Recreate [Feature:Recreate]", func() {
+var _ = framework.SIGDescribe("node", "Recreate [Feature:Recreate]", func() {
```
Same here - it's also debatable if it's not cluster-lifecycle.

ehashman reviewed (commented):
/ok-to-test
/triage accepted

navidshaikh commented on test/e2e/gke_local_ssd.go:
```diff
@@ -29,7 +29,7 @@ import (
 )
 
-var _ = framework.KubeDescribe("GKE local SSD [Feature:GKELocalSSD]", func() {
+var _ = framework.SIGDescribe("storage", "GKE local SSD [Feature:GKELocalSSD]", func() {
```
raised separate PR per sig tag 
 - sig-storage https://github.com/kubernetes/kubernetes/pull/99087

navidshaikh commented:
/retitle Add sig-node tag for e2e tests

aojea commented:
/assign @wojtek-t 

@navidshaikh the job failure is legit, the conformance metadata has changed

navidshaikh commented:
> @navidshaikh the job failure is legit, the conformance metadata has changed

@aojea : Yep, per [conformance test README](https://github.com/kubernetes/kubernetes/blob/master/test/conformance/README.md) I'd need to regenerate the conformance.yaml. However I'm hitting [issue](https://kubernetes.slack.com/archives/C5NV3MT97/p1613377008005800) with bazel. Can we workaround regenerating the conformance.yaml without bazel?

aojea commented:
AFAIK bazel only automates the process, if you are able to get a valid output manually it shouldn't be a problem

navidshaikh commented:
/test pull-kubernetes-integration

navidshaikh commented:
/hold for #99016 to land and follow the refactoring.

wojtek-t commented on test/e2e/examples.go:
```diff
@@ -42,7 +42,7 @@ const (
 )
 
-var _ = framework.KubeDescribe("[Feature:Example]", func() {
+var _ = framework.SIGDescribe("node", "[Feature:Example]", func() {
```
I think that instead of changing anything here, we should probably move this whole file to tests/e2e/node.

@spiffxp

navidshaikh replied:
yes, will update the PR.

navidshaikh replied:
Scoped the PR to only move `test/e2e/examples.go` as the other changes were updating `conformance.yaml`, will raise a separate PR for it.

navidshaikh committed:
test/e2e: Move tests into test/e2e/node package

 - assign sig-node ownership
 - Fix import aliases in test/e2e/node/examples.go
 - Fix merge conflicts in test/e2e/BUILD

Signed-off-by: Navid Shaikh <shaikhnavid14@gmail.com>

navidshaikh commented:
/test pull-kubernetes-node-e2e

navidshaikh commented:
/retest

infra setup error

wojtek-t commented:
/retest

navidshaikh commented:
/retest

 ERROR: (gcloud.compute.addresses.describe) Could not fetch resource:

navidshaikh commented:
/retest

the following error keep popping
```
 ERROR: (gcloud.compute.addresses.describe) Could not fetch resource:
- The resource 'projects/k8s-infra-e2e-boskos-058/regions/us-west1/addresses/bootstrap-e2e-master-ip' was not found
Could not detect Kubernetes master node.  Make sure you've launched a cluster with 'kube-up.sh'
```

aojea commented:
indeed it was failing constantly https://testgrid.k8s.io/presubmits-kubernetes-blocking#pull-kubernetes-node-e2e

aojea commented:
:thinking: 
> W0225 09:40:14.934] F0225 09:40:14.934240    5572 run_remote.go:242] Could not retrieve a image based on image regex "" and family "cos-89-lts": found zero images based on regex "" and family "cos-89-lts" in project "cos-cloud"

aojea commented:
@pacoxu is this https://github.com/kubernetes/kubernetes/pull/98983#issuecomment-785868220 something you have recently fixed with https://github.com/kubernetes/test-infra/pull/21022?

wojtek-t commented:
yes - this was merged couple minutes ago.

wojtek-t commented:
@SergeyKanzhelev - can you please lgtm on behalf of SIG node?

wojtek-t commented:
This LGTM, but I would like someone from SIG-node officially stamp it.

spiffxp commented on test/e2e/node/examples.go:
```diff
@@ -53,17 +53,17 @@ var _ = framework.KubeDescribe("[Feature:Example]", func() {
 	})
 
-	framework.KubeDescribe("Liveness", func() {
+	SIGDescribe("Liveness", func() {
```
SIGDescribe shouldn't be nested, this can be`ginkgo.Describe` (you'll need to import); `[sig-node] [Feature:Example] [sig-node] [Liveness]` seems a bit redundant

spiffxp commented on test/e2e/node/examples.go:
```diff
@@ -110,7 +110,7 @@ var _ = framework.KubeDescribe("[Feature:Example]", func() {
 	})
 
-	framework.KubeDescribe("Secret", func() {
+	SIGDescribe("Secret", func() {
```
Same here, `ginkgo.Describe`

navidshaikh replied:
thanks! fixed (with another one at #L133).

spiffxp reviewed (changes requested):
One thing to change but otherwise LGTM (would like approval from Node though)

spiffxp commented:
/test pull-kubernetes-e2e-kind

spiffxp reviewed (approved):
/approve
Want /lgtm from sig-node

spiffxp commented:
/test pull-kubernetes-e2e-kind

navidshaikh commented:
/test pull-kubernetes-e2e-kind

spiffxp reviewed (approved):
/approve
/lgtm

spiffxp commented:
/hold cancel

SergeyKanzhelev commented:
/lgtm

spiffxp commented:
/test pull-kubernetes-integration
